lib/cards/KCard.vue

```diff
@@ -54,6 +55,40 @@
             :maxLines="titleMaxLines"
           />
         </router-link>
+        <!--
+          Set tabindex to 0 to make title focusable so we
+          can use the same focus ring logic like when title
+          is a router-link. Relatedly set data-focus so that
+          the trackInputModality can set modality to keyboard
+          to make the focus ring display correctly.
+
+          `aria-label` is needed in this mode otherwise some
+          screenreaders don't announce the text in KTextTruncator,
+          for mysterious reasons (likely some internal screenreader logic)
+          -->
+        <span
+          v-else
+          tabindex="0"
+          data-focus="true"
+          class="title"
+          :aria-label="title"
```

Comment:
Here `title` isnt allways necessarily defined, if we use `$slots.title` we wont get the announcement of any title, because this aria-label will be empty, and the slot will be surrounded by an `aria-hidden`. I think we should do an special management here and get the text of the slot to add this as aria-label when the title is being sent as a slot rather than as a prop.

Reply:
Thanks @AlexVelezLl, that's a good catch - I completely forgot there's two title pathways.

> get the text of the slot to add this as aria-label when the title is being sent as a slot rather than as a prop

I thought and played around it and it seems to be tricky. (1) I don't see a straightforward way to obtain it. E.g. for

```
<template #title>
  <span>title</span>
</template>
```

`this.$slots.title`  is an array of `VNode`s and I can't see any attribute on `VNode` that would give the text, neither I've been able to access HTML element. Perhaps you can notice something I missed?

(2) However in any case, any HTML content can be passed via the slot, and I think we can't be 100% assuming it would be suitable `aria-label`.

So I wonder if we should just add `ariaLabel` required prop? For the sake of simplicity and explicitness, we may consider requiring it even when the `title` is passed via a prop.

Reply:
Even though maybe requiring it when `title` prop is there is way too strict and would be an unnecessary hassle. Technically, it's not a problem to make customized check for only when the title slot is used. Will wait to hear from you, perhaps you will have opinion.

Reply:
In KOptionalText we get the slots text in this way: https://github.com/learningequality/kolibri-design-system/blob/39ad629331d74cf90622e86b3a6012b384718fd8/lib/KOptionalText.vue#L54-L74, I think we can combine these approaches, and have an ariaLabel prop, and have this `getSlotsText` as a fallback if we have a slots.title and dont have an aria-label. So for example if we have complex title slots we can use the ariaLabel prop to specifically set the aria-label as needed. But for the most basic use cases like

```
<template #title>
  <span>title</span>
</template>
```

We can live just with the `getSlotsText`.

Reply:
@AlexVelezLl resolved in a way we agreed on Slack